modules/javafx.graphics/src/main/native-glass/monocle/egl/egl_ext.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2011, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+#ifndef __EGL_EXT__
+#define __EGL_EXT__
+#include <jni.h>
+extern long getNativeWindowHandle(const char *v);
+extern long getEGLDisplayHandle();
+extern jboolean doEglInitialize(void* handle);
+extern jboolean doEglBindApi(int api);
+extern jlong doEglChooseConfig (long eglDisplay, int* attribs);
+
+extern jlong doEglCreateWindowSurface (jlong eglDisplay, jlong config,
+     jlong nativeWindow);
+
+extern jlong doEglCreateContext (jlong eglDisplay, jlong config);
+
+extern jboolean doEglMakeCurrent (jlong eglDisplay, jlong drawSurface,
+     jlong readSurface, jlong eglContext);
+
+extern jboolean doEglSwapBuffers(jlong eglDisplay, jlong eglSurface);
```

Comment:
Initially, I wanted to avoid jni.h in this external file. However, since 2 of the functions return a boolean value, I introduced jboolean, but didn't change all values. 
We can still go back to the non-jni types, e.g. by having an int instead of a jboolean, but at this moment I think the presence of jni.h should not be a showstopper. 
Implementations of this interface will be (or bundled with) code that does not know about Java, hence I tried to avoid jni types, but in case jni.h is not available, the implementation can easily typedef the required types.

I made it consistent now to use jlong instead of long.